lighthouse-core/gather/drivers/cri.js

```diff
@@ -16,106 +16,78 @@
  */
 'use strict';
 
-const Driver = require('./driver.js');
-const chromeRemoteInterface = require('chrome-remote-interface');
+const Connection = require('./connection.js');
+const WebSocket = require('ws');
+const http = require('http');
+const hostname = 'localhost';
 const port = process.env.PORT || 9222;
 
-const log = require('../../lib/log.js');
-
-class CriDriver extends Driver {
-  constructor() {
-    super();
-
-    /**
-     * Chrome remote interface instance.
-     */
-    this._cri = null;
-  }
-
+class CriConnection extends Connection {
   /**
-   * @return {!Promise<undefined>}
+   * @override
+   * @return {!Promise}
    */
   connect() {
-    return new Promise((resolve, reject) => {
-      if (this._cri) {
-        return resolve();
-      }
-
-      // Make a new tab, stopping Chrome from accidentally giving CRI an "Other" tab.
-      // Also disable the lint check because CRI uses "New" for the function name.
-      /* eslint-disable new-cap */
-      chromeRemoteInterface.New((err, tab) => {
-        if (err) {
-          log.warn('CRI driver', 'cannot create new tab, will reuse tab.', err);
-        }
-
-        chromeRemoteInterface({port: port, chooseTab: tab}, chrome => {
-          this._tab = tab;
-          this._cri = chrome;
-          // The CRI instance is also an EventEmitter, so use directly for event dispatch.
-          this._eventEmitter = chrome;
-          this.enableRuntimeEvents().then(_ => {
-            resolve();
-          });
-        }).on('error', e => reject(e));
+    return this._runJsonCommand('new').then(response => {
+      const url = response.webSocketDebuggerUrl;
+      return new Promise((resolve, reject) => {
+        const ws = new WebSocket(url);
+        ws.on('open', () => {
+          this._ws = ws;
+          resolve();
+        });
+        ws.on('message', data => this.handleRawMessage(data));
+        ws.on('close', this.dispose.bind(this));
+        ws.on('error', reject);
```

Comment:
once `resolve()` is called above, `reject` here won't do anything (the current version of `connect` has the same problem). I'm not sure the best way to handle this...maybe just disconnect and throw the error?